internal/executor/context/cluster_context.go

```diff
@@ -353,7 +354,10 @@ func (c *KubernetesClusterContext) ProcessPodsToDelete() {
 		}
 
 		if err == nil {
-			err = c.kubernetesClient.CoreV1().Pods(podToDelete.Namespace).Delete(context.Background(), podToDelete.Name, deleteOptions)
+			if podToDelete.DeletionGracePeriodSeconds != nil {
```

Comment:
I don't think this is right.
As I suggested yesterday, I think we should update this function:
https://github.com/G-Research/armada/blob/master/internal/executor/context/cluster_context.go#L432
To look like this:
```
func createDeleteOptions() metav1.DeleteOptions {
	deleteOptions := metav1.DeleteOptions{
                // If left as nil, the default grace period set by the pod is used.
		GracePeriodSeconds: nil,
	}
	return deleteOptions
}
```
Is there a reason you didn't go with this option?

Reply:
this function is used by ingress and service deletion as well so I didn't want to affect those.

Reply:
Good catch. Sounds good.